core/src/main/scala/akka/kafka/internal/ConfigSettings.scala

```diff
@@ -4,32 +4,31 @@
  */
 package akka.kafka.internal
 
-import com.typesafe.config.{Config, ConfigObject, ConfigValue}
+import java.util
+
+import com.typesafe.config.{Config, ConfigObject}
+
+import scala.collection.JavaConverters.asScalaSetConverter
 
 /**
  * INTERNAL API
  */
 private[kafka] object ConfigSettings {
 
   def parseKafkaClientsProperties(config: Config): Map[String, String] = {
-    def collectKeys(c: ConfigObject, prefix: String, keys: Set[String]): Set[String] = {
```

Comment:
@patriknw I didn't faced any issue as such but considering the number of properties we can pass to the producers and consumers I refactored it as a nice to have. WDYT?